Remove -all parameter from Get-AzureRMSubscription and Set-AzureRMContext

yugangw-msft opened the pull request:
@hovsepm @amarzavery 
Please get review started.
I made consolidation on the logics of "get-by-name" and "get-by-id", after confirmed with multiple people that there are duplicates

markcowl commented on src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs:
```diff
@@ -134,15 +134,30 @@ public AzureContext SetCurrentContext(string tenantId)
-                    CreateTenant(tenantId)));
+                TryGetSubscriptionById(tenantId, subscriptionId, out subscription);
+            }
+            else
```
Since this is  a public method, shouldn't we have a check here for subscriptionName not being null or whitespace?  And throw an ArgumentException if not?

markcowl commented on src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs:
```diff
@@ -134,15 +134,30 @@ public AzureContext SetCurrentContext(string tenantId)
         {
-            if (!string.IsNullOrWhiteSpace(tenantId))
+            AzureSubscription subscription;
+            string subscriptionFilter = string.IsNullOrWhiteSpace(subscriptionId) ? subscriptionName : subscriptionId;
```
do we still nees subscriptionFilter?

markcowl commented on src/ResourceManager/Profile/Commands.Profile.Test/AzureRMProfileTests.cs:
```diff
@@ -75,7 +77,7 @@ public void MultipleTenantsAndSubscriptionsSucceed()
             tenantResults = client.ListTenants(DefaultTenant.ToString());
             Assert.Equal(1, tenantResults.Count());
             AzureSubscription subValue;
-            Assert.True(client.TryGetSubscription(DefaultTenant.ToString(), DefaultSubscription.ToString(), out subValue));
```
Can we also cover GetSUbscriptionByName in this test method, or add another unit test for this?

markcowl commented on src/ResourceManager/Profile/Commands.Profile.Test/AzureRMProfileTests.cs:
```diff
@@ -93,7 +96,7 @@ public void SingleTenantAndSubscriptionSucceeds()
             Assert.Equal(1, tenantResults.Count());
             AzureSubscription subValue;
-            Assert.True(client.TryGetSubscription(DefaultTenant.ToString(), DefaultSubscription.ToString(), out subValue));
+            Assert.True(client.TryGetSubscriptionById(DefaultTenant.ToString(), DefaultSubscription.ToString(), out subValue));
```
same

markcowl commented on src/ResourceManager/Profile/Commands.Profile.Test/AzureRMProfileTests.cs:
```diff
@@ -130,7 +135,7 @@ public void NoSubscriptionsInListDoesNotThrow()
             Assert.Equal(0, client.ListSubscriptions().Count());
             AzureSubscription subValue;
-            Assert.True(client.TryGetSubscription(DefaultTenant.ToString(), DefaultSubscription.ToString(), out subValue));
+            Assert.False(client.TryGetSubscriptionById(DefaultTenant.ToString(), DefaultSubscription.ToString(), out subValue));
```
same

markcowl commented:
Reviewed, a couple of nits, otherwise this looks great

yugangw-msft commented:
addressed all review feedback. Waiting for CI passing and the on-demand job passing
http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-on-demand/269/

markcowl commented:
@yugangw-msft Looks like you need to pull the latest
